src/sensors/scripts/sounder_localization.py

```diff
@@ -0,0 +1,110 @@
+#! /usr/bin/env python
+# -*- coding: utf-8 -*-
+# python version 2.7
+
+"""
+	--- Sounder Localization ---
+
+Summary:
+	Package containing the preprocessing of the Sounder location of the AUV, in the
+	case that is underwater. The localization is given with interval analysis.
+
+	Sounder           =>    sounder_localization    =>    location_estimator
+		              |h                            |x
+                                                    |y
+		                                            |z
+	Pressure_Meter    => 
+                      |p
+
+	                  => 
+                      |map
+
+I/O:
+	input message : - std_msgs/Float64
+					- std_msgs/Float64
+					- map
+	output message : geometry_msgs/Vector3
+
+
+# TODO: 
+- do it in C++
+
+"""
+
+
+import rospy
+from std_msgs.msgs import Float64
+from geometry_msgs.msg import Vector3
+#from msgs_pkg.srv import gps_location
+
+
+def listenerSounder(data):
+	""" Listener function which gather Sounder data
+	"""
+	h = data
+	rospy.loginfo("<Sounder>: altitude = %f", h)
+
+
+def listenerPressure(data):
+	""" Listener function which gather Pressure data
+	"""
+	p = data
+	rospy.loginfo("<Pressure>: pressure = %f", p)
+
+
+def evaluateDepth(p):
+	""" Function that computes the depth of the AUV with the pressure
+	"""
+	global RHO, G
+	z = 0  # TODO
+	rospy.loginfo("<Depth>: z = %f", z)
+
+
+if __name__ == '__main__':
+	""" Main program
+	"""
+
+	## ------------- INITIALIZATION -------------
+
+	rospy.init_node('sounder_localization', anonymous=True, log_level=rospy.DEBUG)
+	# In ROS, nodes are uniquely named. If two nodes with the same
+	# node are launched, the previous one is kicked off. The
+	# anonymous=True flag means that rospy will choose a unique
+	# name for our 'listener' node so that multiple listeners can
+	# run simultaneously.
+
+	rate = rospy.Rate(10)  # frenquency in Hertz
+
+	# Server:
+	pub = rospy.Publisher('sounder_position', Vector3, queue_size=10)
+
+
+	# Constants:
+	G = 9.81
+	RHO = 1  # kg/L
+	
+
+	## ------------------ LOOP ------------------ 
+	while not rospy.is_shutdown():
+
+		rospy.Subscriber("pressure", Float64, listenerPressure)
+
+		z = evaluateDepth(p)
+
+		rospy.Subscriber("sounder", Float64, listenerSounder)
```

Comment:
You should remove the subscriber from the loop. 
In ROS a subscriber has to be initialiazed only **once**. It will then automatically call the callback_function when a new message is received.
You can move it before the loop.
(btw, I saw that you often do this in other nodes ([here](https://github.com/EnstaBretagneClubRobo/Cordeliere_ROSws/blob/develop/src/sensors/scripts/gps_localization.py) and [here](https://github.com/EnstaBretagneClubRobo/Cordeliere_ROSws/blob/develop/src/sensors/scripts/imu_localization.py), you should correct it as well)

Reply:
Thanks, I will correct it soon

Reply:
👍